PyRoute/Calculation/TradeCalculation.py

```diff
@@ -228,26 +235,44 @@ def route_update_simple(self, route):
         """
         distance = self.route_distance(route)
 
+        source = route[0]
+        target = route[-1]
+
         # Internal statistics
-        self.galaxy.ranges[route[0]][route[-1]]['actual distance'] = distance
-        self.galaxy.ranges[route[0]][route[-1]]['jumps'] = len(route) - 1
+        rangedata = self.galaxy.ranges[source][target]
+        rangedata['actual distance'] = distance
+        rangedata['jumps'] = len(route) - 1
+
+        if source.is_landmark != target.is_landmark:
+            if source.is_landmark:
+                # landmark = source
+                other = target
+            else:
+                # landmark = target
+                other = source
+
+            self.galaxy.distance_alt_tracking[other] = distance
+
+        self.galaxy.landmarks[(source, target)] = distance
```

Comment:
Rather than building another dict, can we put the distance (or use the distance) calculated in the `self.galaxy.ranges[source][target]` dict?

Reply:
I could, but 3 dict accesses per check, _inside a performance critical region_ , is rarely preferable to a single access that has the same effect.